src/waas/server.py

```
@@ -586,6 +588,50 @@ async def handle_call_tool(name: str, arguments: dict[str, Any]) -> list[types.T
+
+    stdio_server() repoints fd 1 at stderr before server.run() enters this, so
+    connect()'s diagnostics can no longer corrupt the JSON-RPC stream the way
+    they did when it ran at import time. Failures are reported and swallowed:
+    a missing or stale credential must never break the `initialize` handshake,
+    and _dispatch's own `authenticated` guard already answers every tool call
+    with the "Not authenticated" message.
+    """
+    try:
+        if not waas.connect():
+            print("Failed to initialize WAAS connection", flush=True)
+    except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)